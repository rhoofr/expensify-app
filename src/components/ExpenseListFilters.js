import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
} from '../actions/filters';

export class ExpenseListFilters extends Component {
  state = {
    calanderFocused: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    e.target.value === 'date'
      ? this.props.sortByDate()
      : this.props.sortByAmount();
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  render() {
    return (
      <div>
        <input
          type='text'
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value='date'>Date</option>
          <option value='amount'>Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
          startDateId='your_unique_start_date_id' // PropTypes.string.isRequired,
          endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
          endDateId='your_unique_end_date_id' // PropTypes.string.isRequired,
          onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
          focusedInput={this.state.calanderFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
          onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
