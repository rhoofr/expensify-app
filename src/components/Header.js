import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
  return (
    <div>
      <header>
        <h1>Expensify</h1>
        <NavLink activeClassName='is-active' to='/' exact>
          Home
        </NavLink>
        <span> - </span>
        <NavLink activeClassName='is-active' to='/create'>
          Add Expense
        </NavLink>
      </header>
    </div>
  );
};

export default Header;
