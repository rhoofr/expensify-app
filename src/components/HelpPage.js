import React from 'react';

const HelpPage = () => {
  return (
    <div>
      <p>This is from my help page</p>
    </div>
  );
};

export default HelpPage;
