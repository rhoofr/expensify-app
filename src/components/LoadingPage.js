import React from 'react';

export const LoadingPage = () => {
  return (
    <div className='loader'>
      <img
        className='loader__image'
        src='/images/loader.gif'
        alt='Loading...'
      />
    </div>
  );
};

export default LoadingPage;
