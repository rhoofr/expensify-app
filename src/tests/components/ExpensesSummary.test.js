import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with an expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={1234} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal={12344} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render nothing with no expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={0} expensesTotal={0} />
  );
  expect(wrapper).toMatchSnapshot();
});
