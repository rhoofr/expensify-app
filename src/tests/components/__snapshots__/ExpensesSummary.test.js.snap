// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with an expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totaling 
    $12.34
  </h2>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    2
     
    expenses
     totaling 
    $123.44
  </h2>
</div>
`;

exports[`should render nothing with no expenses 1`] = `
<div>
  <h2>
    Viewing 
    0
     
    expenses
     totaling 
    $0.00
  </h2>
</div>
`;
